### PR TITLE
added changes for consent to application and support during case

### PR DESCRIPTION
**Pull request by SasankAchyuta9**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] the keep-helm label has been added, if the helm release should be persisted after a successful build

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/steps/applicant/support-you-need-during-case/SupportYouNeedDuringCaseController.ts [hunk 1]
```diff
@@ -17,60 +19,40 @@ export class SupportYouNeedDuringYourCaseController extends PostController<AnyOb
...
-        if ('C100' === req.session.userCase.caseTypeOfApplication) {
-          req.session.userCase?.applicants?.forEach((applicant: Applicant) => {
-            if (applicant?.value?.user?.idamId === req.session?.user.id) {
-              if (req.url.includes('support-you-need-during-case')) {
-                applicant.value.response['supportYouNeed'] = prepareRequest(req.session.userCase);
-              }
-            }
-          });
+
+        if (partyType === PartyType.APPLICANT) {
```
Comment (vivek-sekhar) [inline src/main/steps/applicant/support-you-need-during-case/SupportYouNeedDuringCaseController.ts:44]: As discussed I told you to use from appSettings to redirect back.

**Review by vivek-sekhar — CHANGES_REQUESTED** [review]

File: src/main/steps/applicant/support-you-need-during-case/SupportYouNeedDuringCaseController.ts (hunk 1 above)
Comment (SasankAchyuta9) [reply src/main/steps/applicant/support-you-need-during-case/SupportYouNeedDuringCaseController.ts]: Yes Vivek, Done this.

File: src/main/steps/applicant/support-you-need-during-case/SupportYouNeedDuringCaseController.ts [hunk 2]
```diff
@@ -17,60 +19,35 @@ export class SupportYouNeedDuringYourCaseController extends PostController<AnyOb
...
-            }
-          });
-        } else {
-          if (req.url.includes('support-you-need-during-case')) {
-            req.session.userCase.applicantsFL401!.response['supportYouNeed'] = prepareRequest(req.session.userCase);
-          }
+        let return_url = RESPONDENT_TASK_LIST_URL;
+        if (partyType === PartyType.APPLICANT) {
+          return_url = APPLICANT_TASK_LIST_URL;
+        } else if (partyType === PartyType.RESPONDENT && req.session.userCase.caseTypeOfApplication === 'C100') {
```
Comment (vivek-sekhar) [inline src/main/steps/applicant/support-you-need-during-case/SupportYouNeedDuringCaseController.ts:41]: why else if party type is required to be checked?
why casetypeofapplication is required?

File: src/main/steps/applicant/support-you-need-during-case/SupportYouNeedDuringCaseController.ts [hunk 3]
```diff
@@ -17,60 +19,35 @@ export class SupportYouNeedDuringYourCaseController extends PostController<AnyOb
...
-              if (req.url.includes('support-you-need-during-case')) {
-                applicant.value.response['supportYouNeed'] = prepareRequest(req.session.userCase);
-              }
-            }
-          });
-        } else {
-          if (req.url.includes('support-you-need-during-case')) {
-            req.session.userCase.applicantsFL401!.response['supportYouNeed'] = prepareRequest(req.session.userCase);
-          }
+        let return_url = RESPONDENT_TASK_LIST_URL;
```
Comment (vivek-sekhar) [inline src/main/steps/applicant/support-you-need-during-case/SupportYouNeedDuringCaseController.ts:38]: why default assignment is required

**Review by vivek-sekhar — CHANGES_REQUESTED** [review]

File: src/main/steps/applicant/support-you-need-during-case/SupportYouNeedDuringCaseController.ts (hunk 3 above)
Comment (SasankAchyuta9) [reply src/main/steps/applicant/support-you-need-during-case/SupportYouNeedDuringCaseController.ts:38]: Yes Removed.

File: src/main/steps/applicant/support-you-need-during-case/SupportYouNeedDuringCaseController.ts (hunk 2 above)
Comment (SasankAchyuta9) [reply src/main/steps/applicant/support-you-need-during-case/SupportYouNeedDuringCaseController.ts:41]: Yes correct , removed.

**Review by vivek-sekhar — APPROVED** [review]

**Pull request merged**